swan-lake/learn-the-language/language-basics.md

```diff
@@ -473,6 +473,98 @@ function nameToString(Name nm) returns string {
+
+For example:
+```ballerina
+function foo(string name) {
+    string:RegExp pattern = re `[a-z]{3}|${name}`;
+}
+```
+In this example, the regular expression pattern includes the ``name`` parameter interpolated into the pattern. This allows for dynamic patterns that can change based on input.
+
+### Unicode property Escape
+
+Unicode general category templates can be used to match characters based on their Unicode properties. These templates can be used with the ``\p{...}`` or ``\P{...}`` escape sequences in a regular expression. Here ``\P`` represents the negation of a unicode template.
+
+Below are some examples of unicode general category templates and their meanings,
+
+- ``\p{L}`` matches any letter 
+- ``\p{Ll}`` matches any lower case letter
+- ``\p{N}`` matches any digit
+- ``\P{P}`` matches any non-punctuation character
+
+Ballerina regex also supports unicode script properties that can be used to match characters based on the language. Inorder to represent a unicode script we have to add ``sc=`` inside the curly braces.
+
+Here are some property meanings,
+
+- ``\p{sc=Arabic}`` matches any character in arabic.
+- ``\p{sc=Latin}`` matches any character in latin.
+
+### Non-capturing groups
+
+Ballerina regexp supports non-capturing groups and flags to control the behavior of regular expression patterns. These allow you to group a set of regular expression constructs without capturing the matched text.
+
+non-capturing groups syntax :=  (?[flags][-][flags]:pattern) 
+
+Ballerina supports four flags that can be used to modify the behavior of regular expression patterns:
+- ``i`` (ignore case) flag: This flag makes the regular expression pattern case-insensitive, so that it matches both uppercase and lowercase letters.
+- ``m`` (multiline) flag: This flag changes the behavior of the ``^`` and ``$`` anchors, so that they match the beginning and end of each line (as opposed to the beginning and end of the entire input string).
+- ``s`` (dot all) flag: This flag changes the behavior of the ``.`` character, so that it matches any character including newline characters.
+- ``x`` (comment) flag: This flag allows you to include comments in your regular expression pattern, using the ``#`` character. Comments are ignored by the regular expression engine.
+
+```ballerina
+// Non-capturing group allows you to apply multiple flags to the pattern inside the parentheses. 
+// `i`, `m`, and `s` flags are applied to the pattern inside the group. 
+// `i` flag makes the pattern case-insensitive.
+// `m` flag makes `^` and `$` match the beginning and end of each line.
+// `s` flag makes the . metacharacters match any character, including line terminators. 
+// This pattern will macth match "hello\nworld", "hello world", and "HELLO \n WORld"
```

Comment:
Why is this a comment instead of a description?

Reply:
This way it is more readable.

Reply:
We use comments in examples, but in these guides we've explained outside the code snippet, in detail. IMO, we need to maintain consistency.